.github/workflows/ci.yml

```diff
@@ -7,15 +7,41 @@ on:
   pull_request:
     branches:
       - master
+  # Allow manually triggering the workflow.
+  workflow_dispatch:
+
+# Cancels all previous workflow runs for the same branch that have not yet completed.
+concurrency:
+  # The concurrency group contains the workflow name and the branch name.
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
 
 jobs:
   tests:
     runs-on: ubuntu-latest
 
+    env: # $BUNDLE_GEMFILE must be set at the job level, so it is set for all steps
+      BUNDLE_GEMFILE: ${{ github.workspace }}/gemfiles/${{ matrix.gemfile }}.gemfile
+
+    # Do not bother building commits with [ci skip] or [skip ci] in the commit message
+    if: "!contains(github.event.commits[0].message, '[ci skip]') && !contains(github.event.commits[0].message, '[skip ci]')"
+
     strategy:
       fail-fast: false
 
       matrix:
+        gemfile:
+          - rails_4_2
+          - rails_5_0
+          - rails_5_1
+          - rails_5_2
+          - rails_6_0
+          - rails_6_1
+          - rails_7_0
+        rubygems:
+          - latest
+        bundler:
+          - latest
```

Comment:
Any reason for this? Link: https://github.com/ruby/setup-ruby/issues/441#issuecomment-1374923371

Reply:
No specific reason. It seems intuitive to use latest versions, as they will have the most bugfixes related to dependency resolution.

But apparently it is not working that way, and it really has nothing to do with this project.

Reply:
The underlying idea is that it normalizes the version as much as possible between builds across many versions of Ruby.  Users of the gem may be keeping pace with rubygems updates in their own builds, and if they aren't nay problems arising from that wouldn't be supported here... It just doesn't make sense to remain stuck on default versions forever.